### PR TITLE
Vision Properties v2

### PR DESCRIPTION
**Pull request by JKL98ISR**
fixed a bug which causes static properties to always use only the first batch
made property calculation to use index
made label/prediction properties not mandatory if given static properties

### CODE REVIEW

File: deepchecks/vision/batch_wrapper.py [hunk 1]
```diff
@@ -103,32 +103,52 @@ def _do_static_prop(self):
...
+            for label in labels:
+                label = label.cpu().detach().numpy()
+                bbox = label[1:]
+                # make sure image is not out of bounds
+                if round(bbox[2]) + min(round(bbox[0]), 0) <= 0 or round(bbox[3]) <= 0 + min(round(bbox[1]), 0):
+                    continue
+                imgs.append(crop_image(img, *bbox))
+        return imgs
+
+    def _get_relevant_data(self, input_type: PropertiesInputType):
```
Comment (nirhutnik) [inline deepchecks/vision/batch_wrapper.py:122]: name of function too general. Maybe add "for properties"?

File: deepchecks/vision/context.py [hunk 2]
```diff
@@ -132,9 +132,9 @@ def __init__(
                                                          [train_properties, test_properties]):
                 if dataset is not None:
                     try:
-                        props = itemgetter(*list(dataset.data_loader.batch_sampler)[0])(properties)
+                        # TODO: add validation
```
Comment (nirhutnik) [inline deepchecks/vision/context.py:135]: Well did you add? :)

File: deepchecks/vision/utils/vision_properties.py [hunk 3]
```diff
@@ -138,6 +139,9 @@ def static_prop_to_cache_format(static_props: STATIC_PROPERTIES_FORMAT) -> PROPE
 
     for input_type in input_types:
         for prop_name in list(static_props[indices[0]][input_type].keys()):
-            props_cache[input_type][prop_name] = [static_props[index][input_type][prop_name] for index in indices]
+            prop_vals = [static_props[index][input_type][prop_name] for index in indices]
+            if input_type == PropertiesInputType.BBOXES.value:
```
Comment (nirhutnik) [inline deepchecks/vision/utils/vision_properties.py:143]: Add an inline comment that explains why for bboxes we flatten the output.

Comment (nirhutnik) [reply deepchecks/vision/utils/vision_properties.py:143]: Also, didn't you change it to "partial_images"?

**Review by nirhutnik — APPROVED** [review]

**Review by ItayGabbay — APPROVED** [review]
